amgcl/detail/qr_solve.hpp

```diff
@@ -0,0 +1,78 @@
+#ifndef AMGCL_DETAIL_QR_SOLVE_HPP_
+#define AMGCL_DETAIL_QR_SOLVE_HPP_
+
+#include <vector>
+#include <cmath>
+
+#include <amgcl/util.hpp>
+#include <amgcl/value_type/interface.hpp>
+#include <amgcl/detail/qr.hpp>
+
+namespace amgcl {
+namespace detail {
+
+enum matrix_shape {
+	tall_matrix,
+	wide_matrix
+};
+
+/// QR solver.
+template <typename value_type, storage_order Order>
+class QRSolve {
```

Comment:
I know I said that we should split the functionality of QR factorization and QR solve, but what I really meant (sorry for misunderstanding) is that the `QR` class should have two public methods: `factorize` and `solve`. `factorize` would do the factorization, and `solve` would call the appropriate form of `factorize`.

So I don't think we need this header and the new class at all.